duffel_api/models/place.py

```diff
@@ -1,32 +1,56 @@
-from ..models import City
+from dataclasses import dataclass
+from typing import Optional, Sequence
 
+from ..models import Airport, City
+from ..utils import get_and_transform
 
+
+@dataclass
 class Place:
     """The city or airport"""
 
+    id: str
+    name: str
+    type: str
+    iata_city_code: Optional[str]
+    iata_country_code: str
+    latitude: Optional[float]
+    longitude: Optional[float]
+    icao_code: Optional[str]
+    time_zone: Optional[str]
+    city_name: Optional[str]
+    city: Optional[City]
+    airports: Optional[Sequence[Airport]]
+
     allowed_types = ["airport", "city"]
 
     class InvalidType(Exception):
         """Invalid type of place"""
 
-    def __init__(self, json):
-        for key in json:
-            value = json[key]
-            if key == "type" and value not in Place.allowed_types:
-                raise Place.InvalidType(value)
-            elif key == "city" and value:
-                value = City(value)
-            elif key == "airports" and value:
-                value = [CityAirport(v) for v in value]
-            setattr(self, key, value)
-
-
-class CityAirport:
-    """The airport associated to a city."""
-
-    def __init__(self, json):
-        for key in json:
-            value = json[key]
-            if key == "city":
-                value = City(value)
-            setattr(self, key, value)
+    def __post_init__(self):
+        if self.type not in Place.allowed_types:
+            raise Place.InvalidType(self.type)
+
+        return self
+
+    @classmethod
+    def from_json(cls, json: dict):
+        """Construct a class instance from a JSON response."""
+        return cls(
+            id=json["id"],
+            name=json["name"],
+            type=json["type"],
+            iata_city_code=json.get("iata_city_code"),
+            iata_country_code=json["iata_country_code"],
+            latitude=json.get("latitude"),
+            longitude=json.get("longitude"),
+            icao_code=json.get("icao_code"),
+            time_zone=json.get("time_zone"),
+            city_name=json.get("city_name"),
+            city=get_and_transform(json, "city", City.from_json),
+            airports=get_and_transform(
```

Comment:
I'd need to do an equivalent for the line above (`city=get_and_transform(json, "city", City.from_json),`), which doesn't lend itself as nicely since it's not a list comprehension?